src/internal/data/mock/collection.go

```diff
@@ -100,3 +100,61 @@ func (c Collection) DoNotMergeItems() bool {
 func (c Collection) FetchItemByName(ctx context.Context, name string) (data.Item, error) {
 	return &Item{}, clues.New("not implemented")
 }
+
+type RestoreCollection struct {
```

Comment:
`BackupCollection` and `RestoreCollection` moved from `src/internal/m365/mock/collection.go`

Reply:
I personally don't think we should fold them together. I think having two separate structs for backup and restore would be useful to keep the tests more readable. That said, if you have a strong opinion on this, I'll merge them.

Reply:
Outside of tests I agree with you.  But inside of tests, the behavior is so similar (and there's no overlaps) I don't see any value in it.  Personal opinion: production code should be as obvious as possible (in terms of ownership, behavior, etc), test code should be as streamlined/centralized as possible (to minimize variations, maintenance, one-off issues, gotchas, etc).  This addresses the most fundamental difficulties with each space: prod code can be difficult to grok (high rate of change, lots of one-off fixes, naming is difficult), test code can be difficult to maintain cleanly (it's easy to let test behavior slip behind prod behavior).

That said, this isn't a hard line.  Follow your gut and we'll be fine.

Reply:
I have folded `BackupCollection` and `RestoreCollection` into Collection. We have to however have a separate `RestoreCollection` with the current implementation as this one takes a data.Collection instead of what the `BackupCollection` was taking.